src/main/java/com/nirmata/workflow/details/internalmodels/StartedTask.java

```diff
@@ -39,6 +41,16 @@ public LocalDateTime getStartDateUtc()
     {
         return startDateUtc;
     }
+    
+    public int getProgress()
+    {
+        return progress;
+    }
+    
+    public void setProgress(int progress)
+    {
+        this.progress = progress;
+    }
 
     @Override
     public boolean equals(Object o)
```

Comment:
I don't think progress field should be used to check whether tasks are equal, or be used to generate hashCode or String. It's just a runtime status of a task, not a personality.